nss-day-cohort-41/nutshell-the-last-rose-of-summer
# Friends section

Description
This PR contains the Friends functionality for Nutshell.

Type of change:
New feature (non-breaking change which adds functionality)

Testing Instructions
     All Primary user friends will be displayed in the 'Friends' section. You may click the 'Friend-' button to unfollow a current friend. In the messages section, clicking the 'Friend+' button will add that user to your following list, display them in an updated Friends section, and change their icon from 'Friend+' to 'Friend following".
     In the 'Add an Item' section, selecting "Friend" will allow the user to search existing users by User Name and successful results will be displayed with the 'Friend+' button, allowing the user to follow them.
     Ensure your prebuilt friends list in JSON matches the format show below in "Optional"

Optional: Paste these friend objects into your friends.database for some starter results
{
      "id": 0,
      "userId": 0,
      "following": 1,
      "date": "2020-06-26T17:58:57.776Z"
    },
    {
      "id": 1,
      "userId": 1,
      "following": 2,
      "date": "2020-06-26T17:58:57.776Z"
    },
    {
      "id": 2,
      "userId": 1,
      "following": 3,
      "date": "2020-06-26T17:58:57.776Z"
    },
    {
      "id": 3,
      "userId": 2,
      "following": 1,
      "date": "2020-06-26T17:58:57.776Z"
    }

Checklist:
 My code follows the style guidelines of this project
 I have performed a self-review of my own code
 My changes generate no new warnings or errors
 I have added test instructions that prove my fix is effective or that my feature works
